### PR TITLE
CI: Fix node tests failure

### LINKED ISSUE #1790: CI: Build LFortran to WASM and Upload Fails Often
Examples:
- https://github.com/lfortran/lfortran/actions/runs/5222312257
- https://github.com/lfortran/lfortran/actions/runs/5218903817
- https://github.com/lfortran/lfortran/actions/runs/5222299646/attempts/1
- https://github.com/lfortran/lfortran/actions/runs/5218730170/attempts/1

The error we receive is `This operation was cancelled`. The previous discussion related to this is here https://github.com/emscripten-core/emsdk/pull/829. It seems that the failure happens because on running `node src/lfortran/tests/test_lfortran.js`, it occupies the whole memory/ram available, there by leading to a termination of the job.

### PR DESCRIPTION
**Pull request by Shaikh-Ubaid**
fixes #1790

### CODE REVIEW

**Comment by Shaikh-Ubaid** [comment]
It seems some checks are being cancelled at the CI. I am unsure what is causing them to cancel.

Is there some limit on the GitHub CI jobs? Did we exhaust some limit (in case it exists)?

File: src/lfortran/tests/CMakeLists.txt [hunk 1]
```diff
@@ -40,6 +40,14 @@ endif()
 # Add one main test suite for LFortran, composed of many individual cpp files:
 add_executable(test_lfortran ${SRC})
 target_link_libraries(test_lfortran lfortran_lib p::doctest)
+if (HAVE_BUILD_TO_WASM)
+    set(WASM_COMPILE_FLAGS "-g0 -fexceptions")
+    set(WASM_LINK_FLAGS
+      "-Oz -g0 -fexceptions -Wall -Wextra -s ASSERTIONS -s ALLOW_MEMORY_GROWTH=1 -s WASM_BIGINT"
```
Comment (Shaikh-Ubaid) [inline src/lfortran/tests/CMakeLists.txt:46]: Providing `-Oz` seems to fix the issue. Other flags are provided just to be extra-safe. I think `-Oz` is more for reducing the size of the generated file. Thus, I am unsure how it fixes the memory utilization. I also experienced this similar behaviour when pursuing https://github.com/lfortran/lfortran/issues/658.

**Comment by Shaikh-Ubaid** [comment]
Jobs at the CI are getting cancelled (for some reason). The CI passed in the 3rd attempt. This is ready.

**Review by certik — APPROVED** [review]
I think this is good!

Thanks @Shaikh-Ubaid for this. Let's merge it and see if this helps.

**Comment by certik** [comment]
I've seen the Windows build being canceled a lot on my PR as well. Maybe the GitHub Actions are unstable today.

**Pull request merged**